tests/Invoke-DbaDiagnosticQuery.Tests.ps1

```diff
@@ -1,16 +1,93 @@
-$CommandName = $MyInvocation.MyCommand.Name.Replace(".Tests.ps1", "")
+# test ouput directory to confirm creation of test files
+$commandname = $MyInvocation.MyCommand.Name.Replace(".Tests.ps1", "")
 Write-Host -Object "Running $PSCommandpath" -ForegroundColor Cyan
-. "$PSScriptRoot\constants.ps1"
+. (Join-path (($PSScriptRoot, '.\' -ne '')[0]) "constants.ps1")  #if running in interactive console, cd to folder and you can run this without error
+write-host "loaded constants"
 
 Describe "$CommandName Integration Tests" -Tags "IntegrationTests" {
-    Context "Verifying output" {
+    BeforeAll {
+        $script:PesterOutputPath = "TestDrive:$commandName"
+
+
+        $database = "dbatoolsci_frk_$(Get-Random)"
+        $database2 = "dbatoolsci_frk_$(Get-Random)"
+        $server = Connect-DbaInstance -SqlInstance $script:instance2
+        $server.Query("CREATE DATABASE $database")
+        $server.Query("CREATE DATABASE $database2")
+
+    }
+    AfterAll {
+        $null = Get-DbaDatabase -SqlInstance $server -Databases @($database, $database2) | Remove-DbaDatabase -Confirm:$false
+        Remove-Item $script:PesterOutputPath -recurse -erroraction silentlycontinue
+
+    }
+    AfterEach {
+        Remove-Item $script:PesterOutputPath -Recurse -erroraction silentlycontinue
+    }
+
+
+    Context "verifying output when running queries" {
         It "runs a specific query" {
-            $results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -QueryName 'Memory Clerk Usage' *>&1
-            $results.Name.Count | Should Be 1
+            $results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -QueryName 'Memory Clerk Usage'
+            @($results).Count | Should -Be 1
```

Comment:
Please use `....Should Be 1`. We do not reference the parameters for Should with `-` prefix so keeps code looking consistent.

Reply:
consistency over anything else but please do mind that pester (from 4.0) is going on with all dashes `-` rather than no dashes at all. New tests should definitely use the `-Be` syntax